git-secrets

```diff
@@ -106,6 +110,48 @@ scan_history() {
   process_output $? "${output}"
 }
 
+# Takes an argument like commitHashA..commitHashB
+# Scans modified files in each commit at each point in history
+scan_between() {
+  # Fail if unstaged changes exist
+  git update-index --refresh
+  [[ -n $(git status --porcelain=v1 2> /dev/null) ]] &&
+    printf "Commit or stash changes first." >&2 && exit 1
+
+  # Require scan range (HEAD/head or SHA1 hash)
+  # Abbreviated hashes minimum 7 chars allowed
+  local -r scan_range="${1}"
+  head_or_sha1="(HEAD|head|[0-9a-f]{7,40})(\^[0-9]{1})?"
+  [[ ! ${scan_range} =~ ^${head_or_sha1}[\.]{2,3}${head_or_sha1}$ ]] &&
+    printf "Must specify a valid scan range, i.e. 'gitHashA..gitHashB'" >&2 && exit 1
+
+  # No patterns means nothing to scan for (OK)
+  local combined_patterns=$(load_combined_patterns)
+  [ -z "${combined_patterns}" ] && return 0
+
+  # No commits means nothing to scan (OK)
+  local -r commits_cmd="git log ${scan_range} --format=format:%h --reverse"
+  IFS=$'\n' read -r -d '' -a commits < <( ${commits_cmd} && printf '\0' )
+  [ -z "${commits}" ] && return 0
+  unset IFS
+
+  local -r current_branch=$(git branch --show-current)
+
+  declare found
+  for commit_hash in "${commits[@]}"; do
+    git checkout "${commit_hash}" >& /dev/null
```

Comment:
Checking out every commit in order to scan here. Would rather a way to scan each commit without checking it out.

Anther safe alternate would be to copy the repo to a temp folder, but it's not realistic with large repos because of the time the copy operation will take.